src/browser/api/system.js

```diff
@@ -610,6 +610,38 @@ exports.System = {
             }
         });
     },
+    getCookies: function(opts, callback, errorCallback) {
+        const { url, name } = opts;
+        if (url && url.length > 0 && name && name.length > 0) {
+            session.defaultSession.cookies.get({ url, name }, (error, cookies) => {
+                if (error) {
+                    log.writeToLog(1, error, true);
+                    errorCallback(`Error getting cookie ${name}`);
+                } else if (cookies.length > 0) {
+                    const data = [];
+                    for (let cookie of cookies) {
+                        if (cookie.httpOnly !== true) {
+                            // excluding value here for security.  Value should be retrieved from document.cookie
+                            data.push({
+                                name: cookie.name,
+                                expirationDate: cookie.expirationDate,
+                                path: cookie.path,
+                                domain: cookie.domain
+                            });
+                        } else {
+                            log.writeToLog(1, `skip cookie ${cookie.name} httpOnly`, true);
+                        }
+                    }
+                    callback(data);
+                } else {
+                    log.writeToLog(1, `cookies result ${cookies.length}`, true);
+                    errorCallback(`Error getting cookie ${name}`);
```

Comment:
Is this really an error situation? If there aren't any cookies by that name, then you get an empty array.

Reply:
Updated as requested